src/language/interpreter.rs

```diff
@@ -63,6 +63,46 @@ where
         + std::ops::Mul<Output = DataType>
         + std::ops::Div<Output = DataType>
         + std::ops::Neg<Output = DataType>
+        + std::ops::Sub<Output = DataType>
+        + std::iter::Sum
+        + num_traits::identities::One
+        + num_traits::identities::Zero
+        + std::cmp::PartialOrd
+        + num_traits::Bounded
+        + Exp
+        + Sqrt
+        + FromNotNanFloat64Literal
+        + ndarray::ScalarOperand,
+    usize: num_traits::cast::AsPrimitive<DataType>,
+{
+    let mut memo_map: std::collections::HashMap<egg::Id, Value<DataType>> = HashMap::default();
```

Comment:
We will likely need to change `Value`'s representation for accesses/tensors, from `ArrayD` to something else...`ArrayViewD` or something like that? Just a note.